src/nlsolve/anderson.jl

```diff
@@ -90,12 +90,12 @@ function (S::NLAnderson{true})(integrator)
   # precalculations
   κtol = κ*tol
 
-  zs = zeros(length(vec(z)), S.n+1)
-  gs = zeros(length(vec(z)), S.n+1)
+  zs = [zero(vec(z)) for i in 1:S.n+1]
+  gs = [zero(vec(z)) for i in 1:S.n+1]
```

Comment:
If possible, move `zs` and `gs` to https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/master/src/nlsolve/type.jl#L3

Reply:
But moving there would make arrays for other methods as well. That would be I guess a lot of memory.

Reply:
Just make them default to `nothing`.

Reply:
As they are arrays, I have default them to `[]` instead.

Reply:
Parameterize it